stdlib/public/runtime/ImageInspectionELF.cpp

```diff
@@ -42,6 +43,62 @@ void record(const swift::MetadataSections *sections) {
 }
 }
 
+#ifndef NDEBUG
+SWIFT_RUNTIME_EXPORT
+const swift::ReflectionInfoDescriptor *swift_getElfReflectionInfo(size_t n) {
+  if (registered == nullptr) {
+    return nullptr;
+  }
+
+  auto selected = registered;
+  while (n-- > 0) {
```

Comment:
We generally avoid post-increment. Can you move the post-increment into the loop and change it to a pre increment?